ipa-core/src/sharding.rs

```diff
@@ -3,9 +3,53 @@ use std::{
     num::TryFromIntError,
 };
 
+use serde::{Deserialize, Serialize};
+
+use crate::helpers::{HelperIdentity, TransportIdentity};
+
+/// This simple trait is used to make aware on what transport dimnsion one is running. Structs like
+/// [`crate::net::client::MpcHelperClient<R>`] use it to know whether they are talking to other
+/// servers as Shards inside a Helper or as a Helper talking to another Helper in a Ring. This
+/// trait can be used to limit the functions exposed by a struct impl depending on the context that
+/// it's being used. Continuing the previous example, the functions a
+/// [`crate::net::client::MpcHelperClient<R>`] provides are dependent on whether it's communicating
+/// with another Shard or another Helper.
+///
+/// This trait is a safety restriction so that structs or traits only expose an API that's
+/// meaningful for their specific context. When used as a generic bound, it also spreads through
+/// the types making it harder to be misused or combining incompatible types, e.g. Using a
+/// [`ShardIndex`] with a [`Ring`].
+pub trait TransportRestriction: Debug + Send + Sync + Clone + 'static {
```

Comment:
this place does not look like a good home for this trait, unless you plan to use it outside of HTTP module?

Reply:
Probably worth consolidating this in `net` module in this case